### PR TITLE
Rvv pre 1.0 2020 09 22

### PR DESCRIPTION
**Pull request by chihminchao**

### CODE REVIEW

File: riscv/processor.cc [hunk 1]
```diff
@@ -767,7 +767,9 @@ void processor_t::set_csr(int which, reg_t val)
 {
 #if defined(RISCV_ENABLE_COMMITLOG)
 #define LOG_CSR(rd) \
-  STATE.log_reg_write[((which) << 4) | 4] = {get_csr(rd), 0};
+  STATE.log_reg_write[((which) << 4) | 4] = { \
+      get_csr(rd, insn_t(0), false, true), \
+      0};
```
Comment (aswaterman) [inline riscv/processor.cc]: I think you should undo this change, and...

File: riscv/processor.h [hunk 2]
```diff
@@ -272,8 +272,8 @@ class processor_t : public abstract_device_t
   void reset();
   void step(size_t n); // run for n cycles
   void set_csr(int which, reg_t val);
-  reg_t get_csr(int which, insn_t insn, bool write);
-  reg_t get_csr(int which) { return get_csr(which, insn_t(0), false); }
+  reg_t get_csr(int which, insn_t insn, bool write, bool peek = 0);
+  reg_t get_csr(int which) { return get_csr(which, insn_t(0), false, false); }
```
Comment (aswaterman) [inline riscv/processor.h]: change this one to pass `true`.  My intent was that `get_csr(int)` is the "peek" form that skips permission checks.

File: riscv/processor.cc (hunk 1 above)
Comment (chihminchao) [reply riscv/processor.cc]: has been changed in push

File: riscv/processor.h (hunk 2 above)
Comment (chihminchao) [reply riscv/processor.h]: has been modified in push

**Comment by aswaterman** [comment]
Thank you!
